packages/langium/src/serializer/json-serializer.ts

```diff
@@ -99,9 +114,11 @@ export class DefaultJsonSerializer implements JsonSerializer {
             } else if (typeof value === 'object' && value !== null) {
                 for (const [name, item] of Object.entries(value)) {
                     if (typeof item === 'object' && item !== null) {
-                        if (isReference(item)) {
-                            const reference = this.linker.buildReference(value as AstNode, name, item.$refNode, item.$refText);
-                            value[name] = reference;
+                        if (isIntermediateReference(item)) {
+                            pathActions.push(() => {
+                                const ref = this.evalRef(root, item.$ref);
+                                value[name] = { ref, $refText: '' };
```

Comment:
Proposal: include the `$refText` in the intermediate references to retain that information. We could add an option to the serialize method to suppress this when it's not required (e.g. when the JSON data is used outside of Langium).